src/components/UserProfiles/UserProfileBase/index.tsx

```diff
@@ -0,0 +1,106 @@
+    logoUrl?: string;
+    avatarUrl?: string;
+}
+
+const UserProfileBase = (props: UserProfileBaseProps) => {
+    const { width, variant, logoUrl, avatarUrl } = props;
+
+    const userProfileLoggedIn: JSX.Element = (
+        <BoxBase width={width}>
+            <BoxBase
+                display="flex"
+                justifyContent="space-between"
+                bgcolor="primary"
+                px={3}
+                pt={3}
+                pb={2}
+            >
+                <BoxBase display="flex" bgcolor="primary">
+                    <ArrowLeftIcon />
+                    <AvatarBase variant="square" src={logoUrl} />
+                    <BoxBase bgcolor="primary" px={1}>
+                        <TypographyBase variant="h6">REPLY</TypographyBase>
+                    </BoxBase>
+                </BoxBase>
+
+                <BoxBase display="flex" justifyContent="flex-end" bgcolor="primary">
+                    <BoxBase bgcolor="primary" px={2}>
+                        <AvatarBase src={avatarUrl} />
+                    </BoxBase>
+                    <SettingsIcon />
+                </BoxBase>
+            </BoxBase>
+            <BoxBase bgcolor="primary" textAlign="center">
+                <ButtonBase
+                    color="secondary"
+                    variant="contained"
+                    startIcon={
+                        <TypographyBase variant="h6">
+                            <EditIcon />
+                        </TypographyBase>
+                    }
+                >
+                    <TypographyBase variant="h5">Compose</TypographyBase>
+                </ButtonBase>
+            </BoxBase>
+        </BoxBase>
+    );
+
+    const userProfileLoggedOut: JSX.Element = (
+        <BoxBase width={width}>
+            <BoxBase display="flex" bgcolor="primary" pt={3} pb={2} px={3}>
+                <ArrowLeftIcon />
+                <AvatarBase variant="square" src={logoUrl} />
+                <BoxBase bgcolor="primary" px={1}>
+                    <TypographyBase variant="h6">REPLY</TypographyBase>
+                </BoxBase>
+            </BoxBase>
+
+            <BoxBase bgcolor="primary" textAlign="center">
+                <TypographyBase variant="body1">WELCOME TO OUR APP</TypographyBase>
+                <BoxBase px={2}>
+                    <DividerBase />
+                </BoxBase>
+                <BoxBase py={2}>
+                    <ButtonBase
+                        color="secondary"
+                        variant="contained"
+                        startIcon={
+                            <TypographyBase variant="h6">
+                                <ExitToAppIcon />
+                            </TypographyBase>
+                        }
+                    >
+                        <TypographyBase variant="h5">Sign In</TypographyBase>
+                    </ButtonBase>
+                </BoxBase>
+            </BoxBase>
+        </BoxBase>
+    );
+

```

Comment:
If you do it like this, userProfileLoggedIn and userProfileLoggedOut will be reinitialized every time this component rerenders, which waste memory. I would advise you to move it out of UserProfileBase but still keep it in the same file